src/screens/CreatePassword/index.tsx

```diff
@@ -1,16 +1,88 @@
-import PasswordCreationForm from '@proapp/components/forms/PasswordCreationForm'
+import { auth } from '@proapp/api/auth'
+import Button from '@proapp/components/common/Button'
+import PasswordCreationForm, { PasswordCreationFormProps } from '@proapp/components/forms/PasswordCreationForm'
 import Welcome from '@proapp/components/Login/Welcome'
+import { getAuthInfo, getAuthRequest } from '@proapp/helpers/auth'
+import { ConsoleLogType, Logger } from '@proapp/helpers/logger'
+import { AuthParamList, AuthScreenProps } from '@proapp/navigation/types'
+import { useLinkTo, useNavigation } from '@react-navigation/native'
+import { NativeStackNavigationProp } from '@react-navigation/native-stack'
 import { Box } from 'native-base'
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import { Platform } from 'react-native'
+import { ActionFeedbackStatus } from '../ActionFeedback'
 
-export const CreatePassword = () => {
+export const CreatePassword = ({ route }: AuthScreenProps<'CreatePassword'>) => {
   const { t } = useTranslation()
+  const navigation = useNavigation<NativeStackNavigationProp<AuthParamList>>()
+  const linkTo = useLinkTo()
+
+  const [expired, setExpired] = useState(false)
+  const [email, setEmail] = useState('')
+  const [formErrors, setFormErrors] = useState<PasswordCreationFormProps['formErrors']>({})
+
+  const token = route?.params?.token as string
+
+  useEffect(() => {
+    if (!token) {
+      Logger(ConsoleLogType.ERROR, { service: 'Create Password', message: 'Token is not defined' })
+      navigation.navigate('LogIn')
+      return
+    }
+
+    const { email, password, exp } = getAuthRequest(token)
+    setEmail(email)
+
+    if (exp < Date.now() / 1000) {
+      Logger(ConsoleLogType.ERROR, { service: 'Create Account', error: new Error(`Create account token has been expired for user ${email}`) })
+      setExpired(true)
+      return
+    }
+
+    auth({ email, password, firstAccess: true })
+      .then((session) => {
+        const firstAccess = getAuthInfo(session?.token)
+        if (firstAccess?.newPasswordRequired === false) {
+          navigation.navigate('LogIn')
+          return
+        }
+        return
```

Comment:
Returns are not needed if they return void, right?

```suggestion
        }
```

Reply:
right